pkg/controller/clusters/status.go

```diff
@@ -214,11 +222,26 @@ func updateWorkersReadyCondition(cluster *anywherev1.Cluster, machineDeployments
 		totalReplicas += int(md.Status.Replicas)
 	}
 
+	totalInplace := 0
+	totalRolling := 0
+	for _, worker := range cluster.Spec.WorkerNodeGroupConfigurations {
+		if worker.UpgradeRolloutStrategy.Type == anywherev1.InPlaceStrategyType {
+			totalInplace++
+		} else {
+			totalRolling++
+		}
+	}
 	// There may be worker nodes that are not up to date yet in the case of a rolling upgrade,
 	// so reflect that on the conditon with an appropriate message.
 	totalOutdated := totalReplicas - totalUpdatedReplicas
 	if totalOutdated > 0 {
-		conditions.MarkFalse(cluster, anywherev1.WorkersReadyCondition, anywherev1.RollingUpgradeInProgress, clusterv1.ConditionSeverityInfo, "Worker nodes not up-to-date yet, %d rolling (%d up to date)", totalReplicas, totalUpdatedReplicas)
+		if totalRolling == totalReplicas {
+			conditions.MarkFalse(cluster, anywherev1.WorkersReadyCondition, anywherev1.RollingUpgradeInProgress, clusterv1.ConditionSeverityInfo, "Worker nodes not up-to-date yet, %d rolling (%d up to date)", totalReplicas, totalUpdatedReplicas)
+		} else if totalInplace == totalReplicas {
+			conditions.MarkFalse(cluster, anywherev1.WorkersReadyCondition, anywherev1.InPlaceUpgradeInProgress, clusterv1.ConditionSeverityInfo, "Worker nodes not up-to-date yet, %d rolling (%d up to date)", totalReplicas, totalUpdatedReplicas)
+		} else {
+			conditions.MarkFalse(cluster, anywherev1.WorkersReadyCondition, anywherev1.UpgradeInProgress, clusterv1.ConditionSeverityInfo, "Worker nodes not up-to-date yet, %d rolling (%d up to date)", totalReplicas, totalUpdatedReplicas)
```

Comment:
We decided to block the ability of having a hybrid configuration for upgrade rollout strategy. It must be the same for CP and all worker groups so we might not need `UpgradeInProgress`